webapp/channels/src/components/admin_console/setting_set.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import SetByEnv from './set_by_env';
+import type {Props as SettingProps} from './setting';
+
+export type Props = Omit<SettingProps, 'nested'>;
+
+export default function SettingSet({
+    children,
```

Comment:
I'm not sure I understand that here. I can understand not passing `children` when we need something more structured (particularly for plugins) or when we would use `cloneElement`, but I don't see why we'd pass something other than `children` here

Reply:
If we have children, it is up to the developer to think "what I want inside". Since these are for settings, what is inside should be quite structured: a setting. Leaving that up to the developer, makes it easier to forget / break things (at least visually).

I know we have history of adding "custom settings" all over the place, and may be not worth it to confine this. But I would love to see us using just "CheckboxSetting", "BooleanSetting", "TextSetting", etc... components, or a "Setting" component with a prop type, that will show different props depending on what type is using.

Reply:
Ah, I see what you mean. I think there's a difference between this and those other settings since this is a layout component for the other `XSetting` components and for any custom settings to use. It's already used by `BooleanSetting` or some `RadioSetting` as well. I'd say that `CheckboxSetting` could use it, but that continues to be a weird case since it's actually multiple config settings under the hood, so I'm not going to go through the headache of unravelling all of that right now.

I think we've actually been successful with most of those components since we do use `BooleanSetting` and stuff for the vast majority of config settings in the System Console. I'd really like to see us do the same thing in the team and user settings as well when we get a chance to